src/Oceananigans.jl

```diff
@@ -162,9 +165,65 @@ const defaults = Defaults()
     AbstractModel
 
 Abstract supertype for models.
+
+Models are required to have the following fields:
+  - `model.clock::Clock`
+  - `model.architecture`
+  - `model.timestepper` with `timestepper.G⁻` and `timestepper.Gⁿ`
 """
 abstract type AbstractModel{TS, A} end
 
+# Method interface for AbstractModel
+
+"""
+    $SIGNATURES
+
+Returns the current iteration of the `model.clock`.
+"""
+iteration(model::AbstractModel) = model.clock.iteration
+
+"""
+    $SIGNATURES
+
+TODO: what does this method do in addition to the constructor?
```

Comment:
ensures consistency between internal variables in the `AbstractModel` and user-specified initial conditions provided by the user through the `set(::AbstractModel; kwargs...)` function

Reply:
Updated in [be79014](https://github.com/CliMA/Oceananigans.jl/pull/4798/commits/be790147d1058a632b8721df37172ae6ba676c8d)